libraries/aws/shadow/src/aws_iot_shadow_operation.c

```diff
@@ -633,12 +633,9 @@ AwsIotShadowError_t _AwsIotShadow_CreateOperation( _shadowOperation_t ** pNewOpe
                 IotLogError( "Failed to create semaphore for waitable Shadow %s.",
                              _pAwsIotShadowOperationNames[ type ] );
 
-                status = AWS_IOT_SHADOW_NO_MEMORY;
-            }
-            else
-            {
                 *pNewOperation = NULL;
                 AwsIotShadow_FreeOperation( pOperation );
+                status = AWS_IOT_SHADOW_NO_MEMORY;
```

Comment:
This does not seem to be the appropriate error status for the failure case of sempaphore creation failure, is it?

Reply:
I think it assumes when IotSemaphore_Create returns false, it implies insufficient memory.

At least FreeRTOS semaphore API works that way. https://www.freertos.org/xSemaphoreCreateBinary.html